container/go/pkg/compat/image.go

```diff
@@ -0,0 +1,147 @@
+/// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Image for intermediate format used in python containerregistry.
+// Uses the go-containerregistry API as backend.
+
+package compat
```

Comment:
Mention that it's mostly from go-containerregistry's layout implementation with modification to understand rules_docker's legacy intermediate format.